src/hooks/useLocalStorage.ts

```diff
@@ -1,30 +1,30 @@
 import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { getLocalStorageItem, setLocalStorageItem } from "@/utils/localStorage";
 
 function useLocalStorage<T>(
   key: string,
   defaultValue: T
-): [T, Dispatch<SetStateAction<T>>] {
-  const [value, setValue] = useState<T>(() => {
-    if (typeof window === "undefined") {
-      return defaultValue;
-    }
+): [T, Dispatch<SetStateAction<T>>];
+
+function useLocalStorage<T>(
+  key: string
+): [T | null, Dispatch<SetStateAction<T | null>>];
 
-    try {
-      const item = window.localStorage.getItem(key);
-      return item ? JSON.parse(item) : defaultValue;
-    } catch (error) {
-      console.error("Error reading localStorage:", error);
-      return defaultValue;
+function useLocalStorage<T>(
+  key: string,
+  defaultValue?: T
+): [T | null, Dispatch<SetStateAction<T | null>>] {
+  const [value, setValue] = useState<T | null>(() => {
+    if (typeof window === "undefined") {
+      return defaultValue ?? null;
     }
+    const storedValue = getLocalStorageItem(key);
+    return storedValue ? storedValue : defaultValue ?? null;
   });
```

Comment:
```suggestion
  const [value, setValue] = useState(() => {
    if (typeof window === "undefined") {
      return defaultValue ?? null;
    }
    const storedValue = getLocalStorageItem<T>(key);
    return storedValue ? storedValue : defaultValue ?? null;
  });
```
`getLocalStorageItem: (key: string) => T | null`이라면 이 부분도 추론될 것 같습니다.